lib/avro/resolution_canonical_form.rb

```diff
@@ -12,20 +14,47 @@ def to_resolution_form(schema)
 
     private
 
-    # TODO: include aliases once the avro Ruby gem supports them
-    # Note: permitted values for defaults are not enforced here.
-    # That should be done at the schema level, and is not done currently
-    # in the Avro Ruby gem
+    def normalize_schema(schema)
+      if schema.type_sym == :bytes && schema.logical_type == DECIMAL_LOGICAL_TYPE
+        add_logical_type(schema, type: schema.type)
+      else
+        super
+      end
+    end
+
     def normalize_field(field)
-      default_value = if field.default?
-                        { default: field.default }
-                      else
-                        {}
-                      end
-      super.merge(default_value)
+      extensions = {}
+      extensions[:default] = field.default if field.default?
+      if field.respond_to?(:aliases) && field.aliases && !field.aliases.empty?
+        extensions[:aliases] = field.aliases.sort
+      end
+
+      super.merge(extensions)
     end
 
-    # TODO: Override normalize_named_type once the Avro Ruby gem supports aliases
-    # def normalized_named_type(schema, attributes = {})
+    def add_logical_type(schema, serialized)
+      if schema.respond_to?(:logical_type) && schema.logical_type == DECIMAL_LOGICAL_TYPE
```

Comment:
Do we need this `respond_to?` check? Looks like that method exists on `Avro::Schema`.

Reply:
Yes, you're correct here too, we can drop it.